File: app/src/main/java/com/enipro/db/dao/FeedDao.java
package com.enipro.db.dao;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;

import com.enipro.data.remote.model.Feed;

import java.util.List;

/**
 * Data Access object used to retrieve information from the sqlite database.
 */

@Dao
public interface FeedDao {

    @Query("SELECT * FROM feeds")
    List<Feed> getFeeds();

    @Query("SELECT * FROM feeds where _id = :feed_id")
    Feed getFeed(String feed_id);

    @Insert
    void insertFeed(Feed feed);

    @Query("DELETE FROM feeds")
    void deleteAll();

    @Delete
    void deleteFeed(Feed feed);

//    @Query("SELECT * FROM feed_comments where _id = :feed_id")
//    List<FeedComment> getFeedComments(String feed_id);
}

File: app/src/main/java/com/enipro/db/dao/FeedDao.kt
package com.enipro.db.dao

import android.arch.paging.DataSource
import android.arch.persistence.room.*

import com.enipro.data.remote.model.Feed

/**
 * Data Access object used to retrieve information from the sqlite database.
 */

@Dao
interface FeedDao {

    @get:Query("SELECT * FROM feeds")
    val feeds: List<Feed>

    @Query("SELECT * FROM feeds where _id = :feed_id")
    fun getFeed(feed_id: String): Feed

    @Insert
    fun insertFeed(feed: Feed)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(feeds: List<Feed>)

    @Query("DELETE FROM feeds")
    fun deleteAll()

    @Delete
    fun deleteFeed(feed: Feed)

    @Query("SELECT * FROM feeds")
    fun selectPagedFeeds(): DataSource.Factory<Int, Feed>
}